tests/algorithms/pytorch/test_pca_algo.py

```diff
@@ -0,0 +1,219 @@
+import logging
+
+import numpy as np
+import pytest
+import torch
+
+from substrafl import execute_experiment
+from substrafl.algorithms.pytorch.torch_fed_pca_algo import TorchFedPCAAlgo
+from substrafl.dependency import Dependency
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.nodes import TestDataNode
+from substrafl.nodes import TrainDataNode
+from substrafl.strategies.fed_pca import FedPCA
+from tests import assets_factory
+from tests import utils
+from tests.utils import download_last_aggregate_model
+
+logger = logging.getLogger(__name__)
+
+
+LINEAR_N_COL = 3
+LINEAR_N_TARGET = 1
+N_EIGENVALUES = 2
+NUM_ROUNDS = 7
+BATCH_SIZE = 1
+
+
+@pytest.fixture(scope="module")
+def torch_pca_algo(numpy_torch_dataset, seed):
+    torch.manual_seed(seed)
+
+    class MyAlgo(TorchFedPCAAlgo):
+        def __init__(
+            self,
+        ):
+            super().__init__(
+                in_features=LINEAR_N_COL,
+                out_features=N_EIGENVALUES,
+                batch_size=BATCH_SIZE,
+                dataset=numpy_torch_dataset,
+            )
+
+    return MyAlgo
+
+
+@pytest.fixture(scope="module")
+def compute_plan(
+    torch_pca_algo,
+    train_linear_nodes_pca,
+    test_linear_nodes_pca,
+    aggregation_node,
+    network,
+    session_dir,
+):
+
+    algo_deps = Dependency(
+        pypi_dependencies=["torch", "numpy"],
+        editable_mode=True,
```

Comment:
It is not clear to me why we need to install torch and numpy in editable mode here, probably worth a comment to explain it.

Reply:
The `editable_mode` option does not refer to the pypi dependencies but to substra, substrafl and substratools.